### PR TITLE
Get Membership Level from Order instead of Request

### PR DESCRIPTION
**Pull request by knit-pay**
Some third-party payment Gateways like Knit Pay don't redirect with Level Id in the URL, due to which addon was incompatible with such payment gateways. Getting memberships level from order instead of request is the safes option because order will be there in all the cases.

### All Submissions:

* [ ] Have you followed the [Contributing guideline](https://github.com/strangerstudios/pmpro-gift-levels/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/pmpro-gift-levels/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Resolves XXX.

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: pmpro-gift-levels.php [hunk 1]
```diff
@@ -149,18 +149,18 @@ function pmprogl_pmpro_after_checkout($user_id)
...
-function pmprogl_pmpro_checkout_before_change_membership_level() {
+function pmprogl_pmpro_checkout_before_change_membership_level( $user_id, $morder ) {
 	global $pmprogl_existing_member_flag, $pmprogl_gift_levels;
 
 	if ( pmpro_hasMembershipLevel()
-	&& ! empty( $_REQUEST['level'] )
+	&& ! empty( $morder->membership_level->id )
 	&& ! empty( $pmprogl_gift_levels ) ) {
-		$checkout_level_id = intval( $_REQUEST['level'] );
+		$checkout_level_id = intval( $morder->membership_level->id );
```
Comment (dparker1005) [inline pmpro-gift-levels.php:163]:
```suggestion
		$checkout_level_id = intval( $morder->membership_id );
```

File: pmpro-gift-levels.php [hunk 2]
```diff
@@ -149,18 +149,18 @@ function pmprogl_pmpro_after_checkout($user_id)
...
 /*
  * Set existing member flags.
  */
-function pmprogl_pmpro_checkout_before_change_membership_level() {
+function pmprogl_pmpro_checkout_before_change_membership_level( $user_id, $morder ) {
 	global $pmprogl_existing_member_flag, $pmprogl_gift_levels;
 
 	if ( pmpro_hasMembershipLevel()
-	&& ! empty( $_REQUEST['level'] )
+	&& ! empty( $morder->membership_level->id )
```
Comment (dparker1005) [inline pmpro-gift-levels.php:161]:
```suggestion
	&& ! empty( $morder->membership_id )
```

File: pmpro-gift-levels.php [hunk 3]
```diff
@@ -72,12 +72,12 @@ function pmprogl_load_plugin_text_domain() {
...
 	
 */
-function pmprogl_pmpro_after_checkout($user_id)
+function pmprogl_pmpro_after_checkout($user_id, $morder)
 {	
 	global $pmprogl_gift_levels, $wpdb, $pmprogl_existing_member_flag;
 	
 	//which level purchased
-	$level_id = intval($_REQUEST['level']);	
+	$level_id = intval($morder->membership_level->id);
```
Comment (dparker1005) [inline pmpro-gift-levels.php:80]:
```suggestion
	$level_id = intval($morder->membership_id);	
```

**Review by dparker1005 — CHANGES_REQUESTED** [review]
Not all orders will have the `$morder->membership_level` property populated, whereas the `$morder->membership_id` property should always be available.

**Comment by dparker1005** [comment]
Only relying on the `$morder` ran into issues if the gift level is free for some reason, so the current solution is to first check the `$_REQUEST` var for the level and, if it can't be found there, to then check the `$morder`.  Let me know if you have any concerns about this and, if so, we can address them before release.

File: pmpro-gift-levels.php [hunk 4]
```diff
@@ -262,7 +262,7 @@ function pmprogl_pmpro_after_checkout($user_id, $morder)
 	global $pmprogl_gift_levels, $wpdb, $pmprogl_existing_member_flag;
 	
 	//which level purchased
-	$level_id = intval($_REQUEST['level']);	
+	$level_id = intval($morder->membership_id);
```
Comment (knit-pay) [inline pmpro-gift-levels.php:265]: this should also be fetched from $morder if $_REQUEST['level'] is not available

Comment (dparker1005) [reply pmpro-gift-levels.php:265]: Here is the current version of that function in the `dev` branch: https://github.com/strangerstudios/pmpro-gift-levels/blob/790052f7a4210997043a2fb79dc84bfb070d0929/includes/checkout.php#L128

We are now always pulling the level from the order, regardless of if the level is available in `$_REQUEST`. Can you please confirm that the current code looks ok?

**Pull request merged**